### PR TITLE
Adapt the example-app to the needs of Data Finland

### PR DESCRIPTION
**Pull request by joakimnordling**

### CODE REVIEW

File: .github/workflows/build-and-deploy.yaml [hunk 1]
```diff
@@ -19,10 +19,11 @@ on:
   push:
     branches:
       - "main"
+      - "dev"
```
Comment (joakimnordling) [inline .github/workflows/build-and-deploy.yaml]: This should be reverted before merging

File: frontend/src/containers/AgentBasicInformation.jsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+export default function AgentBasicInformation() {
+  const [nationalIdentifier, setNationalIdentifier] = useState(
+    DEFAULT_NATIONAL_IDENTIFIERS[DEFAULT_SOURCE]
+  )
+  const [source, setSource] = useState(DEFAULT_SOURCE)
+
+  const [isFetching, setIsFetching] = useState(false)
+  const [isError, setIsError] = useState(false)
+  const [basicInformation, setBasicInformation] = useState({})
```
Comment (lietu) [inline frontend/src/containers/AgentBasicInformation.jsx]:
```suggestion
  const [nationalIdentifier, setNationalIdentifier] = useState(
    DEFAULT_NATIONAL_IDENTIFIERS[DEFAULT_SOURCE]
  )
  const [source, setSource] = useState(DEFAULT_SOURCE)
  const [isFetching, setIsFetching] = useState(false)
  const [isError, setIsError] = useState(false)
  const [basicInformation, setBasicInformation] = useState({})
```

File: frontend/src/containers/AgentBasicInformation.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      resp.data["nationalIdentifier"] = nationalIdentifier
+      setBasicInformation(resp.data)
+      setIsError(false)
+    } else {
+      setIsError(true)
+      setIsFetching(false)
+      throw new Error("Failed to fetch basic information")
+    }
+    setIsFetching(false)
+  }
```
Comment (lietu) [inline frontend/src/containers/AgentBasicInformation.jsx]:
```suggestion
  async function fetchBasicInformation(event) {
    event.preventDefault()
    try {
        setIsFetching(true)
        const resp = await fetchDataProduct(DEFINITION, { nationalIdentifier }, source)
        if (resp.ok) {
          // Add the national identifier to the data; it's not included in the response
          resp.data["nationalIdentifier"] = nationalIdentifier
          setBasicInformation(resp.data)
          setIsError(false)
        } else {
          setIsError(true)
          throw new Error("Failed to fetch basic information")
        }
    } finally {
        setIsFetching(false)
    }
  }
```

JS is weird so not 100% sure if this actually works without testing, but imo worth testing

**Review by lietu — APPROVED** [review]
Added a couple of minor suggestions, but seems good enough

File: frontend/src/containers/AgentBasicInformation.jsx (hunk 2 above)
Comment (lietu) [reply frontend/src/containers/AgentBasicInformation.jsx]: This removes a blank line

File: frontend/src/containers/AgentBasicInformation.jsx (hunk 3 above)
Comment (lietu) [reply frontend/src/containers/AgentBasicInformation.jsx]: This might make the `setIsFetching(false)` slightly more durable

**Review by lietu — APPROVED** [review]

**Pull request merged**